### PR TITLE
Issue #4870: Mark TranslationCheck and JavadocPackageCheck with GlobalStatefulCheck annotation

### PR DESCRIPTION
**Pull request by romani**
Issue #4870

continuation of #5165

### CODE REVIEW

**Comment by romani** [comment]
I sent email to teamcity support to resolve problem.

File: config/import-control.xml [hunk 1]
```diff
@@ -39,8 +39,6 @@
   <allow class="java.lang.annotation.RetentionPolicy" local-only="true"/>
   <allow class="java.lang.annotation.Target" local-only="true"/>
   <allow class="com.puppycrawl.tools.checkstyle.xpath.XpathQueryGenerator" local-only="true"/>
-  <allow class="com.puppycrawl.tools.checkstyle.FileStatefulCheck"/>
-  <allow class="com.puppycrawl.tools.checkstyle.StatelessCheck"/>
```
Comment (rnveach) [inline config/import-control.xml]: Clone service will be in this package, so the imports need to stay.
See https://github.com/checkstyle/checkstyle/pull/4882/files#diff-404a4bfa3acb4cbc7635f4b2f27435b1

File: src/main/java/com/puppycrawl/tools/checkstyle/GlobalStatefulCheck.java [hunk 2]
```diff
@@ -37,7 +37,7 @@
  * Checker guarantees that there will be exactly one check instance
  * This is similar to multi-file validation, which checkstyle does not support fully yet.
  * Please refer to https://github.com/checkstyle/checkstyle/issues/3540 for details.
- * @noinspection AnnotationClass, ClassIndependentOfModule, unused
+ * @noinspection AnnotationClass, ClassIndependentOfModule, unused, ClassOnlyUsedInOnePackage
```
Comment (rnveach) [inline src/main/java/com/puppycrawl/tools/checkstyle/GlobalStatefulCheck.java]: unused and class only in one package should go away, right?

File: src/main/java/com/puppycrawl/tools/checkstyle/checks/javadoc/JavadocPackageCheck.java [hunk 3]
```diff
@@ -32,6 +33,7 @@
  * Checks that all packages have a package documentation. See the documentation
  * for more information.
  */
+@GlobalStatefulCheck
```
Comment (rnveach) [inline src/main/java/com/puppycrawl/tools/checkstyle/checks/javadoc/JavadocPackageCheck.java]: For GlobalStateful it should be fine, but I don't remember how we are going to handle `beginProcessing` for other types of checks.
Each thread will have to create a new instance, so we will have to recall `beginProcessing` each instance?
My own implementation made a true, deep clone before passing it to the thread, so it was never an issue.

Comment (romani) [reply src/main/java/com/puppycrawl/tools/checkstyle/checks/javadoc/JavadocPackageCheck.java]: `beginProcessing` and `finishProcessing` should be executed on each instance of module(in each thread) but before actual multi-thread processing is started.
Lets think about it in other PR/Issues. By this PR I just want to place annotations to other Checks and it is completely ok  to change annotations when we come to actual MT implementation.

So this PR is just declaration of intend, no obligations.

File: src/main/java/com/puppycrawl/tools/checkstyle/GlobalStatefulCheck.java (hunk 2 above)
Comment (romani) [reply src/main/java/com/puppycrawl/tools/checkstyle/GlobalStatefulCheck.java]: eventually yes, IDEA keep complaining on them so for now they should stay.

File: config/import-control.xml (hunk 1 above)
Comment (romani) [reply config/import-control.xml]: we will move them back when it become required.

**Comment by romani** [comment]
@rnveach , ping.

File: src/main/java/com/puppycrawl/tools/checkstyle/checks/javadoc/JavadocPackageCheck.java (hunk 3 above)
Comment (rnveach) [reply src/main/java/com/puppycrawl/tools/checkstyle/checks/javadoc/JavadocPackageCheck.java]: Yes, I am just thinking out loud.

**Pull request merged**